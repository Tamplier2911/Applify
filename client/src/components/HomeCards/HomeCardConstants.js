import {
  IconTimeSVG,
  IconResponsiveSVG,
  IconTechSVG,
  IconShieldSVG,
} from "./HomeCardsStyles";

const homeCardsData = {
  eng: {
    light: [
      {
        id: 201,
        gradient: ["rgb(240, 152, 25)", "rgb(237, 223, 94)"],
        title: "Rapid Development",
        text: "I value your time",
        svg: IconTimeSVG,
      },
      {
        id: 202,
        gradient: ["rgb(72, 128, 236)", "rgb(1, 156, 173)"],
        title: "Responsive Design",
        text: "For any device",
        svg: IconResponsiveSVG,
      },
      {
        id: 203,
        gradient: ["rgb(255, 81, 47)", "rgb(221, 36, 118)"],
        title: "Cutting Edge",
        text: "Supreme technologies",
        svg: IconTechSVG,
      },
      {
        id: 204,
        gradient: ["rgb(60, 165, 92)", "rgb(182, 173, 73)"],
        title: "One Week Warranty",
        text: "Free adjustments",
        svg: IconShieldSVG,
      },
    ],
    dark: [
      {
        id: 201,
        gradient: ["rgb(240, 152, 25)", "rgb(237, 223, 94)"],
        title: "Rapid Development",
        text: "I value your time",
        svg: IconTimeSVG,
      },
      {
        id: 202,
        gradient: ["rgb(72, 128, 236)", "rgb(1, 156, 173)"],
        title: "Responsive Design",
        text: "For any device",
        svg: IconResponsiveSVG,
      },
      {
        id: 203,
        gradient: ["rgb(255, 81, 47)", "rgb(221, 36, 118)"],
        title: "Cutting Edge",
        text: "Supreme technologies",
        svg: IconTechSVG,
      },
      {
        id: 204,
        gradient: ["rgb(60, 165, 92)", "rgb(182, 173, 73)"],
        title: "One Week Warranty",
        text: "Free adjustments",
        svg: IconShieldSVG,
      },
    ],
  },
  rus: {
    light: [
      {
        id: 201,
        gradient: ["rgb(240, 152, 25)", "rgb(237, 223, 94)"],
        title: "Молниеносная Разработка",
        text: "Я ценю ваше время",
        svg: IconTimeSVG,
      },
      {
        id: 202,
        gradient: ["rgb(72, 128, 236)", "rgb(1, 156, 173)"],
        title: "Адаптивный Дизайн",
        text: "Для любого устройства",
        svg: IconResponsiveSVG,
      },
      {
        id: 203,
        gradient: ["rgb(255, 81, 47)", "rgb(221, 36, 118)"],
        title: "Передовые Технологии",
        text: "Лучшее решения",
        svg: IconTechSVG,
      },
      {
        id: 204,
        gradient: ["rgb(60, 165, 92)", "rgb(182, 173, 73)"],
        title: "Одна Неделя Гарантии",
        text: "Бесплатные исправления",
        svg: IconShieldSVG,
      },
    ],
    dark: [
      {
        id: 201,
        gradient: ["rgb(240, 152, 25)", "rgb(237, 223, 94)"],
        title: "Молниеносная Разработка",
        text: "Я ценю ваше время",
        svg: IconTimeSVG,
      },
      {
        id: 202,
        gradient: ["rgb(72, 128, 236)", "rgb(1, 156, 173)"],
        title: "Адаптивный Дизайн",
        text: "Для любого устройства",
        svg: IconResponsiveSVG,
      },
      {
        id: 203,
        gradient: ["rgb(255, 81, 47)", "rgb(221, 36, 118)"],
        title: "Передовые Технологии",
        text: "Лучшее решения",
        svg: IconTechSVG,
      },
      {
        id: 204,
        gradient: ["rgb(60, 165, 92)", "rgb(182, 173, 73)"],
        title: "Одна Неделя Гарантии",
        text: "Бесплатные исправления",
        svg: IconShieldSVG,
      },
    ],
  },
  ukr: {
    light: [
      {
        id: 201,
        gradient: ["rgb(240, 152, 25)", "rgb(237, 223, 94)"],
        title: "Блискавична Розробка",
        text: "Я ціную ваш час",
        svg: IconTimeSVG,
      },
      {
        id: 202,
        gradient: ["rgb(72, 128, 236)", "rgb(1, 156, 173)"],
        title: "Адаптивний Дизайн",
        text: "Для будь-якого пристрою",
        svg: IconResponsiveSVG,
      },
      {
        id: 203,
        gradient: ["rgb(255, 81, 47)", "rgb(221, 36, 118)"],
        title: "Передові Технології",
        text: "Найкраще рішення",
        svg: IconTechSVG,
      },
      {
        id: 204,
        gradient: ["rgb(60, 165, 92)", "rgb(182, 173, 73)"],
        title: "Гарантія Один Тиждень",
        text: "Безкоштовні виправлення",
        svg: IconShieldSVG,
      },
    ],
    dark: [
      {
        id: 201,
        gradient: ["rgb(240, 152, 25)", "rgb(237, 223, 94)"],
        title: "Блискавична Розробка",
        text: "Я ціную ваш час",
        svg: IconTimeSVG,
      },
      {
        id: 202,
        gradient: ["rgb(72, 128, 236)", "rgb(1, 156, 173)"],
        title: "Адаптивний Дизайн",
        text: "Для будь-якого пристрою",
        svg: IconResponsiveSVG,
      },
      {
        id: 203,
        gradient: ["rgb(255, 81, 47)", "rgb(221, 36, 118)"],
        title: "Передові Технології",
        text: "Найкраще рішення",
        svg: IconTechSVG,
      },
      {
        id: 204,
        gradient: ["rgb(60, 165, 92)", "rgb(182, 173, 73)"],
        title: "Гарантія Один Тиждень",
        text: "Безкоштовні виправлення",
        svg: IconShieldSVG,
      },
    ],
  },
};

export default homeCardsData;
