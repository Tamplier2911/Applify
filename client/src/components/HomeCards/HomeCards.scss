.home-cards {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(26.5rem, 1fr));
  grid-gap: 2rem;

  &__card {
    background-color: rgba(216, 182, 255, 0.486);
    min-height: 40rem;
  }
}
