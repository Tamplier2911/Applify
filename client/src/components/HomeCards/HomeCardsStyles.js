import styled, { css } from "styled-components";

import { ReactComponent as TimeSVG } from "../../assets/svg/clock.svg";
import { ReactComponent as ResponsiveSVG } from "../../assets/svg/responsive.svg";
import { ReactComponent as TechSVG } from "../../assets/svg/technologies.svg";
import { ReactComponent as ShieldSVG } from "../../assets/svg/shield.svg";

const svgStyles = css`
  position: absolute;
  width: 6rem;
  height: 6rem;
  display: block;
  object-fit: cover;

  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

export const HomeCardContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(26.5rem, 1fr));
  grid-gap: 2rem;
  padding: 0 0.5rem;
`;

export const IconTimeSVG = styled(TimeSVG)`
  ${svgStyles}
`;

export const IconResponsiveSVG = styled(ResponsiveSVG)`
  ${svgStyles}
`;

export const IconTechSVG = styled(TechSVG)`
  ${svgStyles}
`;

export const IconShieldSVG = styled(ShieldSVG)`
  ${svgStyles}
`;
