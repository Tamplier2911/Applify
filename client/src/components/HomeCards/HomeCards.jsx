// import "./HomeCards.scss";
import React from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";
import { selectCurrentTheme } from "../../redux/theme/theme.selectors";

// components
import Card from "../Card/Card";

// JS Rendering CSS
import { HomeCardContainer } from "./HomeCardsStyles";

// component constats
import homeCardsData from "./HomeCardConstants";

export const HomeCards = ({ lang, theme }) => {
  const currentData = homeCardsData[lang];
  return (
    <HomeCardContainer data-test="home-cards">
      {currentData[theme].map((card) => {
        const { icon, title, text, gradient, svg, id } = card;
        return (
          <Card
            icon={icon}
            title={title}
            text={text}
            gradient={gradient}
            key={id}
            id={id}
            SVG={svg}
          />
        );
      })}
    </HomeCardContainer>
  );
};

const mapStateToProps = () =>
  createStructuredSelector({
    lang: selectCurrentLanguage,
    theme: selectCurrentTheme,
  });

export default connect(mapStateToProps)(HomeCards);
