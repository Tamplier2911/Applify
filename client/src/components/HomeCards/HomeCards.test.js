import React from "react";
import toJson from "enzyme-to-json";

// component
import HomeCards from "./HomeCards";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<HomeCards />", () => {
  const initialStore = {
    theme: {
      currentTheme: "light",
    },
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<HomeCards store={store} />);
  });

  it("renders without an error", () => {
    const homeCards = findByTestAttr(wrapper, "home-cards");
    expect(homeCards.length).toBe(1);
  });

  it("matches snapshot", () => {
    const homeCards = findByTestAttr(wrapper, "home-cards");
    expect(toJson(homeCards)).toMatchSnapshot();
  });
});
