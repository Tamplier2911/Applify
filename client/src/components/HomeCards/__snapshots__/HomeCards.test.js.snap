// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeCards /> matches snapshot 1`] = `ShallowWrapper {}`;
